extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -1793,8 +1793,12 @@ private static BeanInfo findBean(Expression expression, IndexView index,
             return bean;
         } else {
             // User is injecting a non-existing bean
-            incorrectExpressions.produce(new IncorrectExpressionBuildItem(expression.toOriginalString(),
-                    beanName, null, expression.getOrigin()));
+            if (!expression.toOriginalString().endsWith("or(null)")) {
```

Comment:
@mkouba I'm not sure it can be done on the end of expression, what if middle parts are optional, but not the bean. For example, is `ping` bean optional here?

```
package io.quarkus.qute.deployment.inject;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.Optional;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;

import org.jboss.shrinkwrap.api.asset.StringAsset;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.RegisterExtension;

import io.quarkus.qute.Template;
import io.quarkus.test.QuarkusUnitTest;

public class NamedBeanNotFoundSafeTest {

    @RegisterExtension
    static final QuarkusUnitTest config = new QuarkusUnitTest()
            .withApplicationRoot(root -> root
                    .addClass(Ping.class)
                    .addAsResource(new StringAsset("{cdi:ping.pong.val.or(null)}"), "templates/ping.html"));

    @Inject
    Template ping;

    @Test
    public void testTemplate() {
        assertEquals("", ping.render());
    }

    @ApplicationScoped
    //    @Named("ping")
    public static class Ping {

        public Pong pong() {
            return new Pong();
        }

    }

    public static class Pong {

        public Optional<String> val() {
            return Optional.of("pong");
        }

    }

}
```

It passess

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkus.qute.deployment.inject.NamedBeanNotFoundSafeTest
2023-10-17 15:48:01,741 INFO  [io.quarkus] (main) quarkus-qute-deployment 999-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 0.552s. Listening on: http://localhost:8081
2023-10-17 15:48:01,743 INFO  [io.quarkus] (main) Profile test activated. 
2023-10-17 15:48:01,743 INFO  [io.quarkus] (main) Installed features: [cache, cdi, qute, reactive-routes, smallrye-context-propagation, vertx]
2023-10-17 15:48:02,063 INFO  [io.quarkus] (main) quarkus-qute-deployment stopped in 0.021s
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.327 s -- in io.quarkus.qute.deployment.inject.NamedBeanNotFoundSafeTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

Reply:
got it, thanks